### PR TITLE
Fix false critical on OMD backup job when agent runs at the time the backup is about to start

### PR DESCRIPTION
**Pull request by dnlldl**
Prevent this false critical alert:

<html>
<body>
<div style="color: rgb(34, 34, 34); font-family: Arial, Helvetica, sans-serif; font-size: small; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Host | <hostname>
-- | --
Service | OMD <site> backup <name>
Event | OK → CRITICAL
Time | Mon Oct 23 01:30:05 EDT 2023
Summary | Backup completed, it was running for 2 minutes 4 seconds from 2023-10-16 01:30:03 till 2023-10-16 01:32:06, Size: 426 MiB, Next run: 2023-10-23 01:30:00CRIT
Details | Backup completed, it was running for 2 minutes 4 seconds from 2023-10-16 01:30:03 till 2023-10-16 01:32:06Size: 426 MiBNext run: 2023-10-23 01:30:00CRIT
Host Metrics | rta=0.010ms;200.000;500.000;0; pl=0%;80;100;; rtmax=0.038ms;;;; rtmin=0.002ms;;;;
Service Metrics | backup_duration=123.582501;;;; backup_avgspeed=865828.190744;;;; backup_size=446827456;;;;

</div>
</body>
</html>

Basically, this happens when the backup is about to start (here at 01:30:00) but hasn't started yet when the agent checked (around 01:30:00 also in this case but the alert was generated at 01:30:05). In the logs, the backup actually started at 01:30:03, it's normal for a cron job to sometimes have a very small discrepancy, add to that the discrepancy between the check and the time of the alert reported by Checkmk and we get a false critical in this case. The 30 seconds buffer will prevent this corner case from every happening again.

I have read the CLA Document and I hereby sign the CLA or my organization already has a signed CLA.

### CODE REVIEW

**Comment by dnlldl** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/Checkmk/checkmk/blob/master/doc/cla/cla.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> I have read the CLA Document and I hereby sign the CLA or my organization already has a signed CLA.
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

recheck

**Comment by dnlldl** [comment]
I have read the CLA Document and I hereby sign the CLA or my organization already has a signed CLA.

**Comment by dnlldl** [comment]
recheck

**Comment by dnlldl** [comment]
Just curious, is there anything else to do? I'm aware `30 seconds` isn't exactly cute, could just take 2 checks or similar before it turns critical instead.

**Comment by TimotheusBachinger** [comment]
Dear Checkmk Contributor! Unfortunately, we had to re-write our git-repo history, rendering your PR auto-closed. We will therefore rebase your PR onto the current master and reopen it again. Sorry for the inconvenience.

**Comment by tr3pan** [comment]
@TimotheusBachinger I apologize I am writing here. I don't have PR to share, just a comment.

The issue still can happen if the CheckMK check runs delayed more 30 seconds. For example the scheduled backup is at 01:00:00, but the checkmk check for 00:59 happens at 00:59:46 for example. `next_run < time.time() + 30` will be true since the `next_run` time is still from yesterdays backup.

A workaround is to set `Maximum number of check attempts for service` to 2 or higher.

**Comment by TimotheusBachinger** [comment]
Hi @tr3pan I would actually need a new PR on order to get it into our processing queue. If you're interested, feel free to take the current state of this one here.

**Comment by tr3pan** [comment]
@TimotheusBachinger Here it is https://github.com/Checkmk/checkmk/pull/841
